themes/default/content/registry/packages/kubernetes/how-to-guides/managing-resources-with-server-side-apply.md

```diff
@@ -960,3 +960,264 @@ resources:
+behalf, we need to use a transformation to accomplish this.
+
+{{< chooser language "typescript,python,go,csharp" >}}
+
+{{% choosable language typescript %}}
+
+```typescript
+import * as k8s from "@pulumi/kubernetes";
+
+const kruise = new k8s.helm.v3.Chart("kruise", {
+    chart: "kruise",
+    version: "1.3.0",
+    fetchOpts:{
+        repo: "https://openkruise.github.io/charts/",
+    },
+    transformations: [
+        // Ignore changes that will be overwritten by the kruise-manager deployment.
+        (obj: any, opts: pulumi.CustomResourceOptions) => {
+            if (obj.kind === "ValidatingWebhookConfiguration" || obj.kind === "MutatingWebhookConfiguration") {
+                opts.ignoreChanges = [
+                    "metadata.annotations.template",
+                    "webhooks[*].clientConfig",
+                ];
+            }
+        },
+    ],
+});
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```python
+from pulumi_kubernetes.helm.v3 import Chart, ChartOpts, FetchOpts
+
+# Ignore changes that will be overwritten by the kruise-manager deployment.
+def ignore_changes(obj, opts):
+    if obj["kind"] == "ValidatingWebhookConfiguration" or obj["kind"] == "MutatingWebhookConfiguration":
+        opts.ignoreChanges = [
+                    "metadata.annotations.template",
+                    "webhooks[*].clientConfig",
+                ]
+
+
+kruise = Chart(
+    "kruise",
+    ChartOpts(
+        chart="kruise",
+        version="1.3.0",
+        fetch_opts=FetchOpts(
+            repo="https://openkruise.github.io/charts/",
+        ),
+        transformations=[ignore_changes],
+    ),
+)
+```
+
+{{% /choosable %}}
+
+{{% choosable language go %}}
+
+```go
+package main
+
+import (
+    "github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/helm/v3"
+    "github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/yaml"
+    "github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+func main() {
+    pulumi.Run(func(ctx *pulumi.Context) error {
+        _, err := helm.NewChart(ctx, "kruise", helm.ChartArgs{
+            Chart:   pulumi.String("kruise"),
+            Version: pulumi.String("1.3.0"),
+            FetchArgs: helm.FetchArgs{
+                Repo: pulumi.String("https://openkruise.github.io/charts/"),
+            },
+            Transformations: []yaml.Transformation{
+                // Ignore changes that will be overwritten by the kruise-manager deployment.
+                func(state map[string]interface{}, opts ...pulumi.ResourceOption) {
+                    if state["kind"] == "ValidatingWebhookConfiguration" ||
+						state["kind"] == "MutatingWebhookConfiguration" {
+						ignoreChanges := pulumi.IgnoreChanges([]string{
+                            "metadata.annotations.template",
+                            "webhooks[*].clientConfig",
+                        })
+                        opts = append(opts, ignoreChanges)
```

Comment:
This assigns to an argument, which doesn't seem like it would have any effect outside the func.

Reply:
Good catch. It looks like the example that I copied this from may not be working right either. I'm updating these examples now with something that works.